tests/performance_tests/performance_test.py

```diff
@@ -11,203 +12,441 @@
+                if binSearchResults.maxTpsAchieved <= lastMaxTpsAchieved:
+                    break
+            lastMaxTpsAchieved = binSearchResults.maxTpsAchieved
+
+        analysisFinish = datetime.utcnow()
+
+        def calcLocalMax(threadToMaxDict: dict):
+            localMax = 0
+            recThreadCount = 0
+            for threads, tps in threadToMaxDict.items():
+                if tps > localMax:
+                    localMax = tps
+                    recThreadCount = threads
+                else:
+                    break
+            return recThreadCount
+
+        recommendedThreadCount = calcLocalMax(threadToMaxDict=threadToMaxTpsDict)
+
+        return PerformanceTest.PluginThreadOptResult(recommendedThreadCount=recommendedThreadCount, threadToMaxTpsDict=threadToMaxTpsDict,
+                                                     analysisStart=analysisStart, analysisFinish=analysisFinish)
+
+    def createTpsTestReport(self, tpsTestResult: TpsTestResult) -> dict:
+        report = {}
+        report['InitialMaxTpsAchieved'] = tpsTestResult.binSearchResults.maxTpsAchieved
+        report['LongRunningMaxTpsAchieved'] = tpsTestResult.longRunningSearchResults.maxTpsAchieved
+        report['tpsTestStart'] = tpsTestResult.tpsTestStart
+        report['tpsTestFinish'] = tpsTestResult.tpsTestFinish
+        report['InitialSearchResults'] =  {x: asdict(tpsTestResult.binSearchResults.searchResults[x]) for x in range(len(tpsTestResult.binSearchResults.searchResults))}
+        report['InitialMaxTpsReport'] =  tpsTestResult.binSearchResults.maxTpsReport
+        report['LongRunningSearchResults'] =  {x: asdict(tpsTestResult.longRunningSearchResults.searchResults[x]) for x in range(len(tpsTestResult.longRunningSearchResults.searchResults))}
+        report['LongRunningMaxTpsReport'] =  tpsTestResult.longRunningSearchResults.maxTpsReport
+        return report
+
+    def createReport(self,producerThreadResult: PluginThreadOptResult=None, chainThreadResult: PluginThreadOptResult=None, netThreadResult: PluginThreadOptResult=None, tpsTestResult: dict=None) -> dict:
+        report = {}
+        report['perfTestsBegin'] = self.testsStart
+        report['perfTestsFinish'] = self.testsFinish
+        if tpsTestResult is not None:
+            report.update(self.createTpsTestReport(tpsTestResult))
+
+        if producerThreadResult is not None:
+            report['ProducerThreadAnalysis'] = asdict(producerThreadResult)
+
+        if chainThreadResult is not None:
+            report['ChainThreadAnalysis'] = asdict(chainThreadResult)
+
+        if netThreadResult is not None:
+            report['NetThreadAnalysis'] = asdict(netThreadResult)
+
+        report['args'] =  self.prepArgsDict()
+        report['env'] = {'system': system(), 'os': os.name, 'release': release(), 'logical_cpu_count': os.cpu_count()}
+        report['nodeosVersion'] = Utils.getNodeosVersion()
+        return report
+
+    class PtReportEncoder(json.JSONEncoder):
```

Comment:
You didn't want to just import the version logreader has?  `performance_test_basic.py` import from logreader.  I would think `performance_test` could as well, especially if it only import the encoder.

Reply:
Addressed: [c175d6b](https://github.com/AntelopeIO/leap/pull/510/commits/c175d6be97db15670525c6378005e8ca44793d24)